### PR TITLE
Branch find task

### LINKED ISSUE #43: Find and view a task by name
As a user, I can find and view a task by name, so that I can see the details of a task I have saved.

### PR DESCRIPTION
**Pull request by tsh22**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        requireNonNull(model);
+        model.updateFilteredTaskList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_TASKS_LISTED_OVERVIEW, model.getFilteredTaskList().size()));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof FindCommand // instanceof handles nulls
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/logic/commands/FindTaskCommand.java]: FindTaskCommand

File: src/main/java/seedu/address/logic/parser/FindTaskCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import seedu.address.logic.commands.FindTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.TaskNameContainsKeywordsPredicate;
+
+public class FindTaskCommandParser implements Parser<FindTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/logic/parser/FindTaskCommandParser.java]: FindTaskCommand

File: src/main/java/seedu/address/model/ReadOnlyAddressBook.java [hunk 3]
```diff
@@ -14,4 +15,9 @@
      */
     ObservableList<Person> getPersonList();
 
+    /**
+     * Returns an unmodifiable view of the persons list.
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/model/ReadOnlyAddressBook.java]: task

File: src/main/java/seedu/address/model/ReadOnlyAddressBook.java [hunk 4]
```diff
@@ -14,4 +15,9 @@
...
+     * This list will not contain any duplicate persons.
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/model/ReadOnlyAddressBook.java]: tasks

**Review by nicholastanvis — APPROVED** [review]

**Review by lrj689 — APPROVED** [review]

**Review by goatygoatygoat — APPROVED** [review]

**Pull request merged**